packages/data-point-service/lib/revalidation-store.js

```diff
@@ -0,0 +1,85 @@
+const debug = require('debug')('data-point-service:cache')
+const throttle = require('lodash/throttle')
+
+const MAX_STORE_SIZE = 10000
+const THROTTLE_WAIT = 1000 // cleanup in every 1 seconds
+
+/**
+ * It marks and deletes all the keys that have their ttl expired
+ * @param {Map} store Map Object that stores all the cached keys
+ */
+function clear (store) {
+  const forDeletion = []
+  const now = Date.now()
+
+  for (const entry of store) {
+    // mark for deletion entries that have timed out
+    if (now - entry[1][0] > entry[1][1]) {
+      forDeletion.push(entry[0])
+    }
+  }
+
+  if (forDeletion.length > 0) {
+    debug(`local revalidation flags that timed out: ${forDeletion}`)
+  }
+
+  forDeletion.forEach(key => store.delete(key))
```

Comment:
might as well put this inside the conditional block on line 22

Reply:
good call, thanks!